src/views/community/beneficiary/Claim.tsx

```diff
@@ -175,34 +175,39 @@ class Claim extends React.Component<PropsFromRedux & IClaimProps, IClaimState> {
                     isManager === false
                 ) {
                     try {
-                        if (!(await Location.hasServicesEnabledAsync())) {
-                            return;
-                        }
-                        if (
-                            (await Location.getPermissionsAsync()).status !==
-                            'granted'
-                        ) {
-                            return;
+                        const enabled = await Location.hasServicesEnabledAsync();
+                        const permission = await Location.getForegroundPermissionsAsync();
+                        if (!enabled) {
+                            if (
+                                permission.status !==
+                                Location.PermissionStatus.GRANTED
+                            ) {
+                                // if not enabled and previously not allowed, return
+                                return;
```

Comment:
What's the user's feedback after this return?

Reply:
nothing.
what we want is to ask for permission to get the location in case the user has the GPS on, but not allowed on the app. If GPS is off and not allowed, we don't do anything.